R/testGeno.R

```diff
@@ -2,65 +2,42 @@
 ## function that gets an n\times p matrix of p genotypes of n individuals, and a null model, and tests the genotypes associations with the outcomes. 
 ## Genetic data are always assumed complete. 
 ## Types of tests: 
-## Single variant: Wald, score, BinomiRare, interaction. 
+## Single variant: Score, Score.SPA, BinomiRare, interaction. 
 ## Variant set: SKAT, burden, SKAT-O. Multiple types of p-values. Default: Davis with Koenen if does not converge. 
 
 
 # E an environmental variable for optional GxE interaction analysis. 
-testGenoSingleVar <- function(nullmod, G, E = NULL, test = c("Score", "Score.SPA", "Wald"), GxE.return.cov = FALSE){
+testGenoSingleVar <- function(nullmod, G, E = NULL, test = c("Score", "Score.SPA"), GxE.return.cov = FALSE){
     test <- match.arg(test)
 
     G <- .genoAsMatrix(nullmod, G)
 
-    # checks on test 
-    if (test == "Wald" & nullmod$family$family != "gaussian"){
-    	test <- "Score"
-    	message("Cannot use Wald test for non-guassian families, using the score test instead.")
-    }
-    
-    if (test == "Wald" & is.null(E)){
-        Gtilde <- calcGtilde(nullmod, G)
-        res <- .testGenoSingleVarWald(Gtilde, nullmod$Ytilde,
-                                      n=length(nullmod$Ytilde), k=ncol(nullmod$model.matrix))
-    }
-    
-    if (test == "Score" & !is.null(E)){
-    	test <- "Wald"
-    	message("Cannot use Score test for GxE, using the Wald test instead.")
-    }
-        
-    if (test == "Wald" & !is.null(E)){
+    # checks on test
+    if (!is.null(E)){
+        message("Performing GxE test")
         res <- .testGenoSingleVarWaldGxE(nullmod, G, E, GxE.return.cov.mat=GxE.return.cov)
     }
-
+ 
     if(test == "Score.SPA" & nullmod$family$family != "binomial"){
         test <- "Score"
         message("Saddlepoint approximation (SPA) can only be used for binomial family; using Score test instead.")
     }
 
-
     # run the test
     if(test == "Score"){
         Gtilde <- calcGtilde(nullmod, G)
```

Comment:
@smgogarten Is there a reason we kept the `calcGtilde` calculation here instead of in `.testGenoSingleVarScore`? This and the new `RSS0` calculation could each go inside of that function, which would just require `G` and `nullmod` as input then. It would save us from repeating the 2 lines for both Score and Score.SPA.

Also, the new `RSS0` could actually just be computed and saved in the null model, but I'm not sure if that's a bad idea, because it would make the new code incompatible with old null models. What do you think?

Reply:
I think `calcGtilde` is separate because it takes a null model object, whereas the Score/Wald test functions just take matrices and vectors. Theoretically it would testing easier, but I can't actually find anywhere where we test `.testGenoSingleVarScore`directly, so it should not actually matter.

We could set the stage for moving RSS0 in the future by adding it to the null model now, but not expecting it to always be there (`if (is.null(nullmod$RSS0)) <calculate it>`)